src/components/views/projects/new/mdx_project_details_view.tsx

```diff
@@ -9,48 +9,85 @@ import { useDatabase } from '../../../../providers/database_provider'
 
 type MdxProjectViewProps = Record<string, never>
 
 const MdxProjectView: React.FC<MdxProjectViewProps> = () => {
     const db = useDatabase()
-
     const navigate = useNavigate()
+    const [project, setProject] = useState<PouchDB.Core.PutDocument<Project>>()
+
+    useEffect(() => {
+        const fetchAndInitializeProject = async () => {
+            try {
+                const result = await db.allDocs({ include_docs: true })
+                const projects = result.rows
+                    .map(row => row.doc)
+                    .filter(
+                        doc => doc?.type === 'project',
+                    ) as PouchDB.Core.PutDocument<Project>[]
+
+                const validProjects = projects.filter(p => {
+                    return p.data_?.installer?.company_name
+                })
+
+                let installerDefaults: Installer = {
+                    name: '',
+                    company_name: '',
+                    mailing_address: '',
+                    phone: '',
+                    email: '',
+                }
 
-    const [project, setProject] = useState<PouchDB.Core.PutDocument<Project>>(
-        () => {
-            const project = newProject('')
-
-            return {
-                ...project,
-                metadata_: {
-                    ...project.metadata_,
-                    errors: {
-                        data_: {},
-                        metadata_: {
-                            doc_name: [''],
+                if (validProjects.length > 0) {
+                    const mostRecentProject = validProjects.reduce((a, b) => {
+                        return new Date(a.metadata_.last_modified_at) >
+                            new Date(b.metadata_.last_modified_at)
+                            ? a
+                            : b
+                    })
+
+                    installerDefaults = mostRecentProject.data_.installer!
+                }
+
+                const newProjectData = newProject('', undefined, {
+                    installer: installerDefaults,
+                })
+
+                setProject({
+                    ...newProjectData,
+                    metadata_: {
+                        ...newProjectData.metadata_,
+                        errors: {
+                            data_: {},
+                            metadata_: {
+                                doc_name: [''],
+                            },
                         },
                     },
-                },
+                })
+            } catch (error) {
+                console.error('Error fetching projects:', error)
             }
-        },
-    )
+        }
 
-    const isProjectValid = useMemo<boolean>(() => {
-        return !hasErrors(
-            project as PouchDB.Core.Document<Project> & PouchDB.Core.GetMeta,
-        )
-    }, [project])
+        fetchAndInitializeProject()
+    }, [db])
+
+    const isProjectValid = project
```

Comment:
Please wrap this in a call to `useMemo<boolean>` so that the value updates when the `project` changes.